OpenRA.Mods.Common/Traits/Render/DrawLineToTarget.cs

```diff
@@ -72,13 +72,18 @@ public IEnumerable<IRenderable> RenderAfterWorld(WorldRenderer wr)
 			if (targets == null || targets.Count == 0)
 				yield break;
 
+			var validTargets = new List<WPos>();
+			validTargets.Add(self.CenterPosition);
+
 			foreach (var target in targets)
 			{
 				if (target.Type == TargetType.Invalid)
 					continue;
 
-				yield return new TargetLineRenderable(new[] { self.CenterPosition, target.CenterPosition }, c);
+				validTargets.Add(target.CenterPosition);
 			}
+
+			yield return new TargetLineRenderable(validTargets, c);
```

Comment:
Still using yield here, switch for returns, in
   `yield break;`
do
   `return new TargetLineRenderable[0];`
and instead of
   `yield return new TargetLineRenderable(validTargets, c);`
do
   `return new TargetLineRenderable[1] { new TargetLineRenderable(validTargets, c) };`